meilisearch-lib/src/auth_controller/auth_store.rs

```diff
@@ -0,0 +1,257 @@
+                existing
+            }
+            None => false,
+        };
+        wtxn.commit()?;
+
+        Ok(existing)
+    }
+
+    pub fn list_api_keys(&self) -> Result<Vec<Key>> {
+        let mut list = Vec::new();
+        let rtxn = self.env.read_txn()?;
+        for result in self.keys.remap_key_type::<DecodeIgnore>().iter(&rtxn)? {
+            let (_, content) = result?;
+            list.push(content);
+        }
+        Ok(list)
+    }
+
+    pub fn get_expiration_date(
+        &self,
+        key: &[u8],
+        action: Action,
+        index: Option<&[u8]>,
+    ) -> Result<Option<([u8; KEY_ID_LENGTH], DateTime<Utc>)>> {
+        let rtxn = self.env.read_txn()?;
+        match try_split_array_at::<_, KEY_ID_LENGTH>(key) {
+            Some((id, _)) => {
+                let tuple = (&id, &action, index);
+                Ok(self
+                    .action_keyid_index_expiration
+                    .get(&rtxn, &tuple)?
+                    .map(|expiration| (id, expiration)))
+            }
+            None => Ok(None),
+        }
+    }
+
+    pub fn prefix_first_expiration_date(
+        &self,
+        key: &[u8],
+        action: Action,
+    ) -> Result<Option<([u8; KEY_ID_LENGTH], DateTime<Utc>)>> {
+        let rtxn = self.env.read_txn()?;
+        match try_split_array_at::<_, KEY_ID_LENGTH>(key) {
+            Some((id, _)) => {
+                let tuple = (&id, &action, None);
+                Ok(self
+                    .action_keyid_index_expiration
+                    .prefix_iter(&rtxn, &tuple)?
+                    .next()
+                    .transpose()?
+                    .map(|(_, expiration)| (id, expiration)))
+            }
+            None => Ok(None),
+        }
+    }
+
+    fn delete_key_from_inverted_db(
+        &self,
+        wtxn: &mut RwTxn,
+        key: &[u8; KEY_ID_LENGTH],
+    ) -> Result<()> {
+        let mut iter = self
+            .action_keyid_index_expiration
+            .remap_types::<ByteSlice, DecodeIgnore>()
+            .prefix_iter_mut(wtxn, key)?;
+        while let Some(_) = iter.next().transpose()? {
+            // safety: we don't keep references from inside the LMDB database.
+            unsafe { iter.del_current()? };
+        }
+
+        Ok(())
+    }
+}
+
+/// Heed codec allowing to encode/decode everithing that implement Serialize and Deserialize
+/// in order to store it in heed.
+/// This is obviously not the best approach and should never be used for big and numerous objects,
+/// but it is a simple one.
+pub struct SerdeJsonCodec<T>(std::marker::PhantomData<T>);
+
+impl<'a, T> heed::BytesDecode<'a> for SerdeJsonCodec<T>
+where
+    T: Deserialize<'a> + 'a,
+{
+    type DItem = T;
+
+    fn bytes_decode(bytes: &'a [u8]) -> Option<Self::DItem> {
+        serde_json::from_slice(bytes).ok()
+    }
+}
+
+impl<'a, T> heed::BytesEncode<'a> for SerdeJsonCodec<T>
+where
+    T: Serialize + 'a,
```

Comment:
why not using https://docs.rs/heed/0.1.0/heed/types/struct.SerdeJson.html ?

Reply:
Yes, my bad!